scripts/keras_benchmarks/models/lstm_text_generation_benchmark.py

```diff
@@ -0,0 +1,87 @@
+'''
+Original Model from keras/examples
+
+Example script to generate text from Nietzsche's writings.
+At least 20 epochs are required before the generated text
+starts sounding coherent.
+It is recommended to run this script on GPU, as recurrent
+networks are quite computationally intensive.
+If you try this script on new data, make sure your corpus
+has at least ~100k characters. ~1M is better.
+'''
+from __future__ import print_function
+import keras
+from keras.models import Sequential
+from keras.layers import Dense, Activation
+from keras.layers import LSTM
+from keras.optimizers import RMSprop
+from keras.utils import multi_gpu_model
+
+import time
+from model import BenchmarkModel
+from models import timehistory
+if keras.backend.backend() == 'cntk':
+  from gpu_mode import cntk_gpu_mode_config
+from data_generator import generate_text_input_data
+
+class LstmTextGenBenchmark(BenchmarkModel):
+
+    def __init__(self):
+        self._test_name = "lstm_text_generation"
+        self._sample_type="text"
+        self._total_time = 0
+        self._batch_size = 128
+        self._epochs = 2
+        self._num_samples = 1000
+
+    def run_benchmark(self, gpus=0):
+        maxlen = 40
+        char_len = 60
```

Comment:
Just`input_dim` is fine, these are not characters, and the `len` suffix might confuse some readers (semantic collision with sequence length).